### PR TITLE
Add isprime

### PR DESCRIPTION
**Pull request by spdskatr**

### CODE REVIEW

**Comment by spdskatr** [comment]
oops

File: config.example.yml [hunk 1]
```diff
@@ -44,4 +44,4 @@ global_coms:
   - memo
   - remind
   - paste
-  - google
+  - google
```
Comment (Bottersnike) [inline config.example.yml]: Can we have the newline back or was it crashing things?

File: cogs/misc.py [hunk 2]
```diff
@@ -1,178 +1,197 @@
...
+        if num % 5 == 0: 
+            return await ctx.send('Composite. {} literally ends in a 5...'.format(num))
+        if num >= 1000000:
+            return await ctx.send('I may be fast but I\'m not that fast. Try something below 1000000.')
+        a = 3
+        while a <= sqrt(num):
+            if num % a == 0:
+                return await ctx.send('Composite. {0} mod {1} equals 0.'.format(num, a))
+            a = a + (2, 4)[a % 10 == 3]
+        return await ctx.send('Prime! {} is only divisible by 1 and itself'.format(num))
```
Comment (Bottersnike) [inline cogs/misc.py]: Maybe a `.` or `!` at the end of this message?

File: cogs/misc.py [hunk 3]
```diff
@@ -1,178 +1,197 @@
...
-        await ctx.send(die_message)
-
-
-def setup(bot):
-    bot.add_cog(Misc(bot))
+import asyncio
+import base64
+import random
+import time
+from math import sqrt
```
Comment (Bottersnike) [inline cogs/misc.py]: `import math` then `math.sqrt` to avoid polluting the namespace?

File: cogs/misc.py [hunk 4]
```diff
@@ -1,178 +1,197 @@
...
+    @commands.command()
+    async def isprime(self, ctx, num: int):
+        """Very simple prime number checker. Limit is 1000000"""
+        if num in [2, 3, 5, 7]:
+            return await ctx.send("*sigh* Yes, {} is prime.".format(num))
+        if num % 2 == 0:
+            return await ctx.send('Why would you think {}, an even number, is prime!?'.format(num))
+        if num % 5 == 0: 
+            return await ctx.send('Composite. {} literally ends in a 5...'.format(num))
+        if num >= 1000000:
```
Comment (Bottersnike) [inline cogs/misc.py]: A _bit_ pedantic, but `>=` is redundant here as `==` is caught by `num % 5` earlier in the code. (Just use `>`)

**Review by Bottersnike — CHANGES_REQUESTED** [review]
Other than these, it looks pretty good.

**Comment by AlexCMueller** [comment]
There are only about 70,000 primes in this range, so a lookup table is kinda feasable. The primality checking in general is pretty crappy, has anyone actually tested to see how long it would take for noah's computer to check 999,983?

EDIT: proposal doesn't work, only a full lookup table would lead to significant speedup

**Comment by Bottersnike** [comment]
On mediocre hardware, `10**10` will execute in around `0.3175048660126221` seconds. Upping it to `10**11` will take it to `1.0105299202432678`s, but I think `10**6` is probably a bit too small as that can easily execute in `0.0027852597500475137`s

**Comment by AlexCMueller** [comment]
Botter, do you actually understand how primes work?

Those numbers are basically best case scenarios. They get caught in the fist n%2 check and get passed. This is why I suggested  999983 as a test case, as it's worse case (and more representative of larger numbers)

**Comment by Bottersnike** [comment]
Heh.. Derp..

**Comment by Bottersnike** [comment]
`999983` still had an average of `0.00011714474831082188` seconds. Oh, and those other numbers were wrong because they were the totals for all of the iterations not the average (because `timeit` is that good).

**Comment by AlexCMueller** [comment]
I forget how fast modern computers are, I suppose we could go to 10**7 but I'd rather stay in the range of integers.

**Review by spdskatr — COMMENTED** [review]
All good, 1 million was just an arbitrary limit for the time being. I'll change it

**Comment by spdskatr** [comment]
After a bit of testing the prime number 1000000005721 yielded 0.47870827575113567 seconds

**Comment by spdskatr** [comment]
made changes

**Pull request merged**